src/main/java/seedu/duke/model/gates/Gate.java

```diff
@@ -1,17 +1,40 @@
 package seedu.duke.model.gates;
 
+import seedu.duke.DukeException;
+
 public abstract class Gate {
     protected int output;
     protected int input;
+    protected int secondInput;
```

Comment:
I see, will have to change the class diagrams for this.

Reply:
Oof sozz should've informed you